lib/travis/build/addons/jwt.rb

```diff
@@ -0,0 +1,43 @@
+require 'travis/build/addons/base'
+require 'jwt'
+
+module Travis
+  module Build
+    class Addons
+      class Jwt < Base
+        SUPER_USER_SAFE = true
+
+        def before_before_script
+          secrets = nil
+          if config.is_a?(String)
+            secrets = [config]
+          elsif config.is_a?(Array)
+            secrets = config
+          else
+            secrets = config.values
```

Comment:
Okay I'll look into it more.

Does supporting strings at all make sense? I'm not super comfortable with
writing Travis yaml files so I'm still learning as I go.

On Jul 26, 2016 9:15 AM, "Hiro Asari" <notifications@github.com> wrote:

> In lib/travis/build/addons/jwt.rb
> <https://github.com/travis-ci/travis-build/pull/356#discussion_r72283212>:
>
> > +require 'jwt'
> > +
> > +module Travis
> > +  module Build
> > +    class Addons
> > +      class Jwt < Base
> > +        SUPER_USER_SAFE = true
> > +
> > +        def before_before_script
> > +          secrets = nil
> > +          if config.is_a?(String)
> > +            secrets = [config]
> > +          elsif config.is_a?(Array)
> > +            secrets = config
> > +          else
> > +            secrets = config.values
>
> I am not sure if this should be a desirable default.
>
> As I mentioned previously, I do not think config should be a Hash at all,
> since the keys are discarded.
>
> With that in mind, this portion can be simplified to:
>
> secrets = Array(config)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/travis-ci/travis-build/pull/356/files/b5a1fb6de21ee237ca552a761d612899bf11ee0f#r72283212>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGuB7yxXDm9uutYLTAAErRalbzDuE-fks5qZjJvgaJpZM4DHeYE>
> .
>